### PR TITLE
Add scoped query filter rules and remove Twitter's t

### PR DESCRIPTION
**Pull request by fmarier**
Resolves brave/brave-browser#26756

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Important: You must **clear all local data** (or use a fresh profile) before running these tests. The Twitter URL will only work correctly the first time (known issue: brave/brave-browser#23742).

1. Open <https://www.instagram.com/desusbloodyshoes/?igshid=abcdef> and confirm that the `igshid` parameter is missing from the query string.
2. Open <https://brave.com/?igshid=abcdef> and confirm that the `igshid` parameter is **present** in the query string.
3. Open <https://twitter.com/fmarier/status/1574869891978784769?s=21&t=lkfjeslkfjes&fbclid=1234> and confirm that neither the `t` nor the `fbclid` parameters are in the URL (but `s=21` should still be there).

### CODE REVIEW

**Review by iefremov — APPROVED** [review]

**Review by arthuredelstein — APPROVED** [review]

File: browser/net/brave_site_hacks_network_delegate_helper_unittest.cc [hunk 1]
```diff
@@ -251,6 +251,9 @@ TEST(BraveSiteHacksNetworkDelegateHelperTest, QueryStringFiltered) {
        {"https://example.com/?fbclid=1&a+b+c=some%20thing&1%202=3+4",
         "https://example.com/?a+b+c=some%20thing&1%202=3+4"},
        // Conditional query parameter stripping
+       {"https://example.com/?igshid=1234", ""},
```
Comment (arthuredelstein) [inline browser/net/brave_site_hacks_network_delegate_helper_unittest.cc:254]: Why is the second string empty here? (I might just be missing something.)

File: browser/net/brave_site_hacks_network_delegate_helper_unittest.cc [hunk 2]
```diff
@@ -251,6 +251,9 @@ TEST(BraveSiteHacksNetworkDelegateHelperTest, QueryStringFiltered) {
...
+       {"https://www.instagram.com/?igshid=1234", "https://www.instagram.com/"},
+       {"https://example.com/?mkt_tok=123&foo=bar&mkt_unsubscribe=1", ""},
```
Comment (arthuredelstein) [inline browser/net/brave_site_hacks_network_delegate_helper_unittest.cc:256]: Also this one?

File: browser/net/brave_site_hacks_network_delegate_helper_unittest.cc (hunk 1 above)
Comment (arthuredelstein) [reply browser/net/brave_site_hacks_network_delegate_helper_unittest.cc:254]: Got it. Thanks!

File: browser/net/brave_site_hacks_network_delegate_helper_unittest.cc (hunk 2 above)
Comment (fmarier) [reply browser/net/brave_site_hacks_network_delegate_helper_unittest.cc:256]: (same reason)

**Pull request merged**